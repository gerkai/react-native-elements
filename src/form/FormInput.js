import PropTypes from 'prop-types';
import React, { Component } from 'react';
import {
  TextInput,
  StyleSheet,
  View,
  Platform,
  Dimensions,
  Text as NativeText,
} from 'react-native';
import colors from '../config/colors';
import normalize from '../helpers/normalizeText';

const { width } = Dimensions.get('window');

class FormInput extends Component {
  getRef = () => {
    return this.props.textInputRef
      ? this.refs[this.props.textInputRef]
      : this.input;
  };

  focus() {
    this.getRef() && this.getRef().focus();
  }

  blur() {
    this.getRef() && this.getRef().blur();
  }

  clearText() {
    this.getRef() && this.getRef().clear();
  }

  render() {
    const {
      containerStyle,
      inputStyle,
      textInputRef,
      containerRef,
      selectionColor,
      ...attributes
    } = this.props;
    return (
      <View
        ref={containerRef}
        style={[styles.container, containerStyle && containerStyle]}
      >
        <TextInput
          ref={textInputRef || (input => this.input = input)}
          selectionColor={selectionColor || colors.grey3}
          style={[styles.input, inputStyle && inputStyle]}
          {...attributes}
        />
      </View>
    );
  }
}

FormInput.propTypes = {
  containerStyle: View.propTypes.style,
  inputStyle: NativeText.propTypes.style,
  selectionColor: PropTypes.string,
  textInputRef: PropTypes.string,
  containerRef: PropTypes.string,
};

const styles = StyleSheet.create({
  container: {
    marginLeft: 15,
    marginRight: 15,
    ...Platform.select({
      ios: {
        borderBottomColor: colors.grey4,
        borderBottomWidth: 1,
        marginLeft: 20,
        marginRight: 20,
      },
    }),
  },
  input: {
    ...Platform.select({
      android: {
        height: 46,
      },
      ios: {
        height: 36,
      },
    }),
    width: width,
    color: colors.grey3,
    fontSize: normalize(14),
  },
});

export default FormInput;
