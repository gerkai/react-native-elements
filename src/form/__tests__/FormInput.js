import React from 'react';
import {shallow} from 'enzyme';
import FormInput from '../FormInput';

describe('FormInput Component', () => {
  it('should render without issues', () => {
    const component = shallow(<FormInput />);

    expect(component.length).toBe(1);
    expect(component).toMatchSnapshot();
  });

  it('should have a TextInput component', () => {
    const component = shallow(<FormInput />);

    expect(component.find('TextInput').length).toBe(1)
  });

  it('should call onChangeText function on adding new value', () => {
    const customFunction = jest.fn();
    const component = shallow(<FormInput onChangeText={customFunction} />);
    component.find('TextInput').simulate('ChangeText', 'Text Change')

    expect(customFunction).toHaveBeenCalled();
  });
});
