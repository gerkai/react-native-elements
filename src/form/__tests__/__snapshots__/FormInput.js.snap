// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput Component should render without issues 1`] = `
<AnimatedComponent
  style={
    Array [
      Object {
        "borderBottomColor": "#bdc6cf",
        "borderBottomWidth": 1,
        "marginLeft": 20,
        "marginRight": 20,
      },
      undefined,
      Object {
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      },
    ]
  }
>
  <TextInput
    style={
      Array [
        Object {
          "color": "#86939e",
          "fontSize": 17.5,
          "minHeight": 36,
          "width": 750,
        },
        undefined,
      ]
    }
  />
</AnimatedComponent>
`;
